pkg/controller/controller_binding.go

```diff
@@ -1439,21 +1450,17 @@ func (c *controller) pollServiceBinding(binding *v1beta1.ServiceBinding) error {
 
 			setServiceBindingCondition(
 				binding,
-				v1beta1.ServiceBindingConditionReady,
-				v1beta1.ConditionFalse,
+				v1beta1.ServiceBindingConditionFailed,
```

Comment:
This will make the behavior inconsistent between sync and async binding. In the sync case, if there is an error injecting the binding, the reconciliation is retried until the retry duration is exceeded. In the async case, if there is an error injecting the binding, the reconciler immediately sets the state of the binding to Failed. One way or another, the two scenarios need to be consistent.

Reply:
I think the synchronous behavior is the ideal one. I would like to rework the logic to be able to retry the fetching and injection after we know the operation succeeded, but would like to do that in a follow-up. I think it's okay to have the slight inconsistency for the time being.

Reply:
OK. I can accept that.